controller/src/main/java/io/pravega/controller/server/rpc/auth/StrongPasswordProcessor.java

```diff
@@ -0,0 +1,101 @@
+ *  is digested with the retrieved iterations and salt. The generated digest is then cross checked against the created digest.
+ */
+public class StrongPasswordProcessor {
+
+    private final int iterations = 1000;
+
+    /*
+     * @param password              The incoming password.
+     * @param encryptedPassword     The stored password digest.
+     * @return                      true if the password matches, false otherwise.
+     * @throws NoSuchAlgorithmException encryption exceptions.
+     * @throws InvalidKeySpecException encryption exceptions.
+     */
+    public boolean checkPassword(String password, String encryptedPassword) throws NoSuchAlgorithmException, InvalidKeySpecException {
+        String storedPassword = new String(fromHex(encryptedPassword));
+        String[] parts = storedPassword.split(":");
+        int iterations = Integer.parseInt(parts[0]);
+        byte[] salt = fromHex(parts[1]);
+        byte[] hash = fromHex(parts[2]);
+
+        PBEKeySpec spec = new PBEKeySpec(password.toCharArray(), salt, iterations, hash.length * 8);
+        SecretKeyFactory skf = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA1");
+        byte[] testHash = skf.generateSecret(spec).getEncoded();
+
+        int diff = hash.length ^ testHash.length;
+        for (int i = 0; i < hash.length && i < testHash.length; i++) {
+            diff |= hash[i] ^ testHash[i];
+        }
+        return diff == 0;
+    }
+
+    /*
+     * @param userPassword The password to be encrypted.
+     * @return The encrypted string digest.
+     * @throws NoSuchAlgorithmException encryption exceptions.
+     * @throws InvalidKeySpecException encryption exceptions.
+     */
+    public String encryptPassword(String userPassword) throws NoSuchAlgorithmException, InvalidKeySpecException {
+        char[] chars = userPassword.toCharArray();
+        byte[] salt = getSalt();
+
+        PBEKeySpec spec = new PBEKeySpec(chars, salt, iterations, 64 * 8);
+        SecretKeyFactory skf = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA1");
+        byte[] hash = skf.generateSecret(spec).getEncoded();
+        return toHex((iterations + ":" + toHex(salt) + ":" + toHex(hash)).getBytes());
+    }
+
+    private byte[] getSalt() throws NoSuchAlgorithmException {
+        SecureRandom sr = SecureRandom.getInstance("SHA1PRNG");
+        byte[] salt = new byte[16];
+        sr.nextBytes(salt);
+        return salt;
+    }
+
+    private String toHex(byte[] array) throws NoSuchAlgorithmException {
+        BigInteger bi = new BigInteger(1, array);
+        String hex = bi.toString(16);
+        int paddingLength = (array.length * 2) - hex.length();
+        if (paddingLength > 0) {
+            return String.format("%0"  +paddingLength + "d", 0) + hex;
+        } else {
+            return hex;
+        }
+    }
+
+    private byte[] fromHex(String hex) {
+        byte[] bytes = new byte[hex.length() / 2];
```

Comment:
I think you want: https://docs.oracle.com/javase/7/docs/api/java/math/BigInteger.html#BigInteger(java.lang.String,%20int)